### DISCUSSION TITLE
UI Changes

### DISCUSSION THREAD

**Discussion by ghost** [ludwig-v/wireless-carplay-dongle-reverse-engineering#37] [Q&A]
Has anyone tried to change the GUI? I am working on Apple CarPlay theme for the UI.

I have managed to do a few things: 

1) Correct English Text
2) Remove useless screen Elements: random icons in the bottom left that don’t actually do anything, small vertical line
3) Remove the bottom horizontal line
4) Change the Red icons to Car Play Style Blue Icons 
5) Theme using Car Play Icons - Back/Search/Settings/Find and Menu background screens with rounded corners matching the “dark mode” car play theme.
6) Bluetooth Scanning Animation - MacOS Spinning Wheel 

Still Pending:
1) Rearrange Elements on screen
2) Change Main Backgrounds to Grey
3) Edit the Settings Menu.

Issues: 
———————
SETTINGS:

There is no way to edit the options in the list box in Settings screen in the ui.tar.gz - 

Does anyone know where I can edit the setting menu? It doesn’t seem to be in ui.tar.gz

**Comment by dmavromatis** [comment]
Hi! I modified my settings screen to match my clean minimal look. I changed the settings background from the gradient to black.

From what I have seen, you cannot change or remove the items in the list.  Those are built dynamically and populated into the
`[::s(Theme0_SetupList_CarlinKit)]` element.

As for the QR code question I read else where, that is dynamically added as well, you can comment out the following:
`[::s(Theme0_CMDIQRCode_Carlinkit)]desx=center;desy=center;cx=90;cy=90;[/]`

**Reply by ghost** [reply]
I am looking to actually modify the text and list items in settings. 

It has to come from somewhere, we just need to find the file holding the text. 

Thank you for the code sniped to remove the QR. 

These config files are messy, no consistency when coding, some lines are compressed into a single line some are 4 short lines.

**Comment by dmavromatis** [comment]
From what I have seen poking around, the list items are dynamically populated via the code in the firmware -- it's always in English.  Just so we're both on the same page -- you're talking about the "Setup" page, accessed by tapping on the gear.  The only thing you can modify is the "Setup" string (1030) and the layout of the page... not the items in the list.  Same is true tying to get access to the paired device names listbox... those are also populated via the firmware.  If you figure out a way to access them please post here.

**Reply by ghost** [reply]
Correct. 

I will check.

**Reply by dmavromatis** [reply]
The only thing I can find is if you want to just show the BT Device name (without button bg) somewhere on the AutoKit version, you can probably use this: `[::mybtname(Theme0_BTNameTxt)]txtcolor=<0xff,0xff,0xff>;bold=2;italic=1; desy=122*full/480;cx=full;cy=39*full/480;txtsize=32*full/800; txt="...";[/]`

**Reply by ghost** [reply]
Mine looks different than yours? 

[::s(Theme0_BTNameTxt_CarlinKit)]

Why does yours have? mybtname?

**Reply by dmavromatis** [reply]
Yes, sorry -- I was just giving an example -- that line is from the previous layout (non-Autokit).  You can see how adding in the other elements turns it into a button (bg).

**Comment by kentherzz** [comment]
@IcedLatte5, do you have picture of the UI you made?

**Reply by ghost** [reply]
Hi,

I am not finished. I so far have built all the images. This has been on going for the past few days in trial and error.

I just started working on rearranging and resizing elements now. Their code is pretty messy. 

 Couple things on the todo list: 

-modify the Font .otf to use the Apple Car play offical font. I had already tried to straight swap the .otf but that did not work. So the next step is to replace the basic a-z0-9 characters manually in the font. 

-rearrange all elements on screen

**Reply by kentherzz** [reply]
Ok, nice. Can we just replace the ui.tar.gz with custom one with U2W.sh. How did you do this?
I want to modify translation files and I believe they are locatedd in the ui.tar.gz, no?

**Reply by ghost** [reply]
Yes. 

Create a backup of the ui.tar.gz 

You will need to unzip it, make changes to the language file and then tar it again. 

If you make any mistakes you will be booted to a black screen when you plug in the dongle in the car. 

You can go back fix your mistakes or upload the original ui.tar.gz file to revert. 

I tried to use 7zip to retar but that failed. You will need to properly tar it using Mac or Cygwin on Windows. 

Also, to elaborate - you will need to instal the new package that allows you to SFTP to the device then you can just Use a tool like WinSCP to connect to 192.168.50.2 and copy over the modified file. 

It’s a bit of a learning curve.

**Reply by kentherzz** [reply]
Thanks, my plan was to use U2W.sh script to replace ui.tar.gz, but of course much easier and intuitive with SSH.
